patch/patcher.go

```diff
@@ -531,12 +531,50 @@ func (p *Patcher) patchIdent(id *ast.Ident, obj types.Object) {
 			if v.Tag == nil {
 				v.Tag = &ast.BasicLit{}
 			}
-			v.Tag.Value = "`" + strings.TrimSpace(strings.Trim(v.Tag.Value, "` ")+" "+tags) + "`"
+
+			v.Tag.Value = mergeTags(v.Tag.Value, tags)
 			log.Printf("Add tags:\t%q.%s %s", obj.Pkg().Path(), id.Name, v.Tag.Value)
 		}
 	}
 }
 
+// Override tags
+func mergeTags(oldTag, newTag string) string {
+	oldTags := strings.Split(strings.TrimSpace(strings.Trim(oldTag, "`")), " ")
+	var tagKeys []string
+	var kvMap = make(map[string]string)
+	for _, tag := range oldTags {
+		if kv := strings.Split(tag, ":"); len(kv) > 0 {
+			tagKeys = append(tagKeys, kv[0])
+			if len(kv) > 1 {
+				kvMap[kv[0]] = kv[1]
+			}
+		}
+	}
+	newTags := strings.Split(strings.TrimSpace(strings.Trim(newTag, "`")), " ")
+	for _, tag := range newTags {
+		if kv := strings.Split(tag, ":"); len(kv) > 0 {
+			if _, ok := kvMap[kv[0]]; !ok {
```

Comment:
This will break if the check is skipped on line 549 due to an empty tag value.

Either always put a blank value in `kvMap` for a present tag name, or omit the tag name from `tagKeys` if the value is blank or nonexistent.

Reply:
the `oldTag` generated by grpc,so it won't be empty.
if  `newTag` value is empty,It will be skipped.

Reply:
Regardless, I think this tool shouldn’t make assumptions about the output of grpc.